src/main/java/org/opensearch/geospatial/ip2geo/action/GetDatasourceTransportAction.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geospatial.ip2geo.action;
+
+import java.util.List;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.geospatial.ip2geo.common.DatasourceFacade;
+import org.opensearch.geospatial.ip2geo.jobscheduler.Datasource;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+/**
+ * Transport action to get datasource
+ */
+@Log4j2
+public class GetDatasourceTransportAction extends HandledTransportAction<GetDatasourceRequest, GetDatasourceResponse> {
+    private final DatasourceFacade datasourceFacade;
+
+    /**
+     * Default constructor
+     * @param transportService the transport service
+     * @param actionFilters the action filters
+     * @param datasourceFacade the datasource facade
+     */
+    @Inject
+    public GetDatasourceTransportAction(
+        final TransportService transportService,
+        final ActionFilters actionFilters,
+        final DatasourceFacade datasourceFacade
+    ) {
+        super(GetDatasourceAction.NAME, transportService, actionFilters, GetDatasourceRequest::new);
+        this.datasourceFacade = datasourceFacade;
+    }
+
+    @Override
+    protected void doExecute(final Task task, final GetDatasourceRequest request, final ActionListener<GetDatasourceResponse> listener) {
+        if (request.getNames().length == 0 || (request.getNames().length == 1 && "_all".equals(request.getNames()[0]))) {
```

Comment:
1. check for nulls on the getNames().
2. Why we are seeing _all only at first index. Should we just check at all the places.
3. We can abstract this whole condition to a separate function for better readability and extension in future.